strawberry/promise/promise_list.py

```diff
@@ -0,0 +1,165 @@
+# Copy of the Promise library (https://github.com/syrusakbary/promise) with some
```

Comment:
Should this file (and the others from the promise lib) be marked excluded from the coverage suite?

Reply:
Also, can we get a comment/mini readme to what "some modifications" means?

Reply:
I wonder, I guess in future we might change this code a lot (ie updating the code to only use py 3 things), maybe we can change say "originally based on promises"